test/behavior/optional.zig

```diff
@@ -498,6 +498,20 @@ test "optional of noreturn used with orelse" {
     try expect(val == 123);
 }
 
+test "mutable optional of noreturn" {
+    if (builtin.zig_backend == .stage2_sparc64) return error.SkipZigTest; // TODO
+
+    var a: ?noreturn = null;
+    if (a) |*ptr| {
+        _ = ptr;
+        @compileError("bad");
```

Comment:
should this test go in `test/cases/` since this this test failing results in a compile error rather than a runtime error?

Reply:
No, there are plenty of behavior tests which will trigger compile errors if they fail.